reconcile/utils/ocm/search_filters.py

```diff
@@ -0,0 +1,231 @@
+
+    def render(self, key: str) -> str:
+        escaped = self.value.replace("'", "''")
+        return f"{key}='{escaped}'"
+
+
+@dataclass
+class StringLikeCondition:
+    """
+    string equality OCM search filter condition
+    """
+
+    value: str
+
+    def render(self, key: str) -> str:
+        return f"{key} like '{self.value}'"
+
+
+@dataclass
+class ListInCondition:
+
+    values: list[Any]
+
+    def render(self, key: str) -> str:
+        quoted_values = ",".join(map(lambda x: f"'{x}'", self.values))
+        return f"{key} in ({quoted_values})"
+
+
+@dataclass
+class OrCondition:
+
+    filters: list["Filter"]
+
+    def render(self, _: str) -> str:
+        conditions = " or ".join(map(lambda f: f.render(), self.filters))
+        return f"({conditions})"
+
+
+@dataclass
+class DateRangeCondition:
+
+    start: Optional[Union[datetime, str]]
+    end: Optional[Union[datetime, str]]
+
+    def render(self, key: str) -> str:
+        conditions = []
+        resolved_start = self.resolve_start()
+        resolved_end = self.resolve_end()
+        if resolved_start and resolved_end and resolved_end < resolved_start:
+            raise InvalidFilterError(
+                f"Invalid date range: start={resolved_start.isoformat()} end={resolved_end.isoformat()}"
+            )
+        if resolved_start:
+            conditions.append(f"{key} >= '{resolved_start.isoformat()}'")
+        if resolved_end:
+            conditions.append(f"{key} <= '{resolved_end.isoformat()}'")
+        return " and ".join(conditions)
+
+    def resolve_start(self) -> Optional[datetime]:
+        return DateRangeCondition.resolve_date(self.start) if self.start else None
+
+    def resolve_end(self) -> Optional[datetime]:
+        return DateRangeCondition.resolve_date(self.end) if self.end else None
+
+    @staticmethod
+    def resolve_date(date: Union[datetime, str]) -> datetime:
+        if isinstance(date, datetime):
+            return date
+        parsed = dateparser.parse(
+            date,
+            settings={"RELATIVE_BASE": DateRangeCondition.now()},
+        )
+        if parsed is None:
+            raise InvalidFilterError(f"Invalid relative date: {date}")
+
+        return parsed
+
+    @staticmethod
+    def now() -> datetime:
+        return datetime.now(tz=timezone.utc)
+
+
+class InvalidFilterError(Exception):
+    pass
+
+
+class InvalidChunkRequest(Exception):
+    pass
+
+
+class Filter:
+    def __init__(self, conditions: Optional[dict[str, FilterCondition]] = None):
+        self.conditions: dict[str, FilterCondition] = conditions or {}
+
+    def copy(self) -> "Filter":
+        conditions_copy = self.conditions.copy()
+        return Filter(conditions_copy)
+
+    def add_condition(self, key: str, condition: FilterCondition) -> "Filter":
+        copied = self.copy()
+        copied.conditions[key] = condition
+        return copied
+
+    def eq(self, key: str, value: Optional[str]) -> "Filter":
+        if value:
```

Comment:
Please add doc strings to all important public methods, like `eq`, `like`, `is_in` ...